ul/src/pdu/writer.rs

```diff
@@ -0,0 +1,745 @@
+use crate::error::Result;
+use crate::pdu::*;
+use byteordered::byteorder::{BigEndian, WriteBytesExt};
+use std::io::Write;
+
+fn write_n(writer: &mut dyn Write, bytes_to_write: &[u8]) -> Result<()> {
```

Comment:
This is a similar concern to that of `read_n`. You can use [`Write::write_all`](https://doc.rust-lang.org/std/io/trait.Write.html#method.write_all) instead, right?